ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/api/useFetchReports.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-void */
+import { useCallback, useEffect, useState } from 'react';
+
+import {
+    fetchReportConfigurations,
+    fetchReportStatus,
+    fetchReportLastRunStatus,
+} from 'services/ReportsService';
+import { ReportStatus } from 'services/ReportsService.types';
+import { Report } from '../types';
+import { getErrorMessage } from '../errorUtils';
+
+type Result = {
+    reports: Report[];
+    isLoading: boolean;
+    error: string | null;
+};
+
+type FetchReportsResult = {
+    fetchReports: () => void;
+} & Result;
+
+const defaultResult = {
+    reports: [],
+    isLoading: false,
+    error: null,
+};
+
+function useFetchReports(): FetchReportsResult {
+    const [result, setResult] = useState<Result>(defaultResult);
+
+    const fetchReports = useCallback(async () => {
+        setResult({
+            reports: [],
+            isLoading: true,
+            error: null,
+        });
+
+        try {
+            const reportConfigurations = await fetchReportConfigurations();
+            const reports: Report[] = await Promise.all(
+                reportConfigurations.map(async (reportConfiguration): Promise<Report> => {
+                    // @TODO: The API returns a 500 when there's no report status. For now we'll do a try/catch, but
+                    // we should wait for backend to change this to a 400
```

Comment:
I understand why a 500 is a bad return status, but 400 usually means "you made a bad request" outside of the API contract.

I don't think that's what you mean, or what the API should return just because there is no report status.

Did you mistype, or do we need to clarify this more with the API team?